### PR TITLE
ZLIB_INCLUDE_DIR has not been added as an include dir.

### PR DESCRIPTION
**Pull request by l-korous**

### CODE REVIEW

File: cmake/configure/configure_1_zlib.cmake [hunk 1]
```diff
@@ -18,3 +18,4 @@
 #
 
 CONFIGURE_FEATURE(ZLIB)
+INCLUDE_DIRECTORIES(${ZLIB_INCLUDE_DIR})
```
Comment (bangerth) [inline cmake/configure/configure_1_zlib.cmake]: you need an endline at the end of the file.

**Comment by bangerth** [comment]
This seems reasonable to me (other than the missing endline and the accidental deletion of the three files). But @tamiko might want to have a look.

File: cmake/configure/configure_1_zlib.cmake (hunk 1 above)
Comment (l-korous) [inline cmake/configure/configure_1_zlib.cmake]: Done.

**Comment by tamiko** [comment]
No, this is not correct. Please do not use any global ```INCLUDE_DIRECTORIES```.

Further, there is already the following snippet of code in ```cmake/modules/FindZLIB.cmake```:
```
DEAL_II_PACKAGE_HANDLE(ZLIB
  LIBRARIES REQUIRED ZLIB_LIBRARY
  INCLUDE_DIRS REQUIRED ZLIB_INCLUDE_DIR
  CLEAR ZLIB_INCLUDE_DIR ZLIB_LIBRARY
  )
```
And this works as expected: ```detailed.log```:
```
#        DEAL_II_WITH_ZLIB set up with external dependencies
#            ZLIB_VERSION = 1.2.8
#            ZLIB_INCLUDE_DIRS = /usr/include
#            ZLIB_LIBRARIES = /usr/lib64/libz.so
```
Can you please paste your ```detailed.log``` and the compilation error you get?

**Comment by tamiko** [comment]
(Best with complete compiler invocation.)

**Comment by tamiko** [comment]
@l-korous *ping*

**Comment by l-korous** [comment]
Hi, sorry to be late. Matthias - I'm not stupid :-), here is your 'working' detailed log for me:
#        DEAL_II_WITH_ZLIB set up with external dependencies
#            ZLIB_VERSION = 1.2.3
#            ZLIB_INCLUDE_DIRS = c:/hpfem/agros-windows-dependencies/32/include
#            ZLIB_LIBRARIES = c:/hpfem/agros-windows-dependencies/32/lib/zlib.lib

Here is my directory 'c:/hpfem/agros-windows-dependencies/32/include':
...
zlib.h
...

And here is the compiler error:
Error C1083: Cannot open include file: 'zlib.h': No such file or directory
In file: C:\hpfem\dealii\bundled\boost-1.56.0\libs\iostreams\src\zlib.cpp

The compiler invocation is rather complicated in Visual Studio, but I can plainly see, that there is no such thing for which a gcc alternative would be '-Ic:/hpfem/agros-windows-dependencies/32/include'. When I add the 'wrong' INCLUDE_DIRECTORIES, it works, because such a thing (actually called in MSVC 'Additional Include Directories') gets included.

**Comment by luca-heltai** [comment]
Is it possible that the problem is a missing 

`USER_INCLUDE_DIRS REQUIRED ZLIB_DIRS` 

in the `FindZLIB.cmake`? I had similar problems in the past with OpenCascade...

**Comment by tamiko** [comment]
It seems that I slowly forget how I set up the build system :-D

> I'm not stupid :-)

I never meant to imply any such thing.

I objected because I do not want to have global include directories to be set. The correct fix is to use the include statement [1] at the correct (local) scope. I'll open a pull request in a minute - please test.

[1] I believed I had refactored everything to not use ```INCLUDE_DIRECTORIES``` any more - I originally only kept those statements for compatibility with 1.8.7.

**Pull request closed without merging**